### DISCUSSION TITLE
How to disable window minimizing in Qtile?

### DISCUSSION THREAD

**Discussion by developer1105** [qtile/qtile#3410] [Q&A]
In my 'rc.lua' file (from awesome wm) there is this code:
```
-- Disable window minimizing
client.connect_signal("property::minimized", function(c)
    c.minimized = false
end)
```
Is there something similar to this in Qtile? As soon as the user minimizes the window, the window manager un-minimizes it.

**Comment by m-col** [comment]
You can do something similar with hooks:
```py
from libqtile import hook

@hook.subscribe.focus_change
def _(win):
    if win.minimized:
        win.cmd_toggle_minimize()
```

**Reply by developer1105** [reply]
It did not work...
Maybe parameter 'win' points to the next window after focus change? (Instead of the previous one that was minimized)

**Reply by elParaguayo** [reply]
Yeah, that's probably what happens.

How about something like:
``` python
from libqtile import hook, qtile

@hook.subscribe.focus_change
def _(win):
    last_focused = getattr(qtile, "last_focused", False)
    if last_focused and last_focused.minimized:
        last_focused.cmd_toggle_minimize()
    qtile.last_focused = win
```

**Reply by m-col** [reply]
My bad, the hook doesn't receive any arguments: https://github.com/qtile/qtile/blob/master/libqtile/backend/x11/window.py#L1326

You could still use it to scan the windows in the current group and un-minimize the minimized ones.

This all aside, why is your window becoming minimized? What application is it?

**Reply by developer1105** [reply]
There are some apps that display maximize/minimize buttons even though qtile removes title bar. (Examples: naver-whale and opera-developer). The TaskList widget allows one to test it with any app.

**Reply by developer1105** [reply]
Unfortunately elParaguayo's code didn't work as well...
One thing I have seen in many qtile config files are settings for floating windows.
Maybe there could be something like that to handle this issue:
```
    never_minimize_rules=[
        Match(wm_class="naver-whale"),
        Match(wm_class="opera-developer"),
    ]
```

**Reply by m-col** [reply]
Does setting `auto_minimize = False` in your config work?

**Reply by HackerTheFox** [reply]
I know this is an old issue, but I just wanted to add if someone comes across this. variable does work. Its located at the bottom of the qtile's config file.
`auto_minimize = False`

**Comment by developer1105** [comment]
Unfortunately "auto_minimize = False" did not work.

I used to minimize the windows myself. I saw the minimize button and then I thought I was on a regular window manager, and I just clicked on it, inadvertently.

The command below works only in the shell:
```
qtile cmd-obj -o group 2 -f unminimize_all
```

These two lines worked only with keybindings
```
lazy.group.unminimize_all()

qtile.cmd_unminimize_all()
```

I have tested all the possible hooks, but no one worked.

**Reply by developer1105** [reply]
Just updating the list of browsers that have this "issue"...
(For experienced users it is not a problem, but for newcomers it might be annoying to see the window getting "lost" after an accidental minimizing.)

```
    never_minimize_rules=[
        Match(wm_class="naver-whale"),
        Match(wm_class="opera-developer"),
        Match(wm_class="dissenter-browser"),
    ]
```
